pkg/helper/helpers.go

```diff
@@ -198,6 +203,25 @@ func DeleteAppliedResources(resources []workapiv1.AppliedManifestResourceMeta, r
 			continue
 		}
 
+		// If resource is not owned by the appliedManifestWork, ignore it
+		if !IsOwnedBy(owner, u.GetOwnerReferences()) {
+			continue
+		}
+
+		// If there are multiple owners, remove ours and skip
+		remainigOwners := RemoveOwner(owner, u.GetOwnerReferences())
+		if len(remainigOwners) > 0 {
```

Comment:
nit, factoring like this would have helped me.

```go
if currOwners  := u.GetOwnerReferences(); currOwners > 1{
  remainingOwners := RemoveOwner(owner, u.GetOwnerReferences())

  // do stuff
  continue
}
```